pkg/build/api/validation/validation.go

```diff
@@ -752,3 +732,56 @@ func ValidateNodeSelector(nodeSelector map[string]string, fldPath *field.Path) f
 	}
 	return allErrs
 }
+
+func diffBuildSpec(newer, older buildapi.BuildSpec) (string, error) {
+	newerObj := &buildapi.Build{Spec: newer}
+	olderObj := &buildapi.Build{Spec: older}
+	diffBytes, err := CreateBuildPatch(olderObj, newerObj)
+	if err != nil {
+		return "", err
+	}
+	return string(diffBytes), nil
+}
+
+func CreateBuildPatch(older, newer *buildapi.Build) ([]byte, error) {
+	codec := kapi.Codecs.LegacyCodec(v1.LegacySchemeGroupVersion)
+
+	newerJSON, err := runtime.Encode(codec, newer)
+	if err != nil {
+		return nil, fmt.Errorf("error encoding newer: %v", err)
+	}
+	olderJSON, err := runtime.Encode(codec, older)
+	if err != nil {
+		return nil, fmt.Errorf("error encoding older: %v", err)
+	}
+	patch, err := strategicpatch.CreateTwoWayMergePatch(olderJSON, newerJSON, &v1.Build{})
```

Comment:
random passing comment, not asking for a change - I wonder if by using json patch (rfc6902) it would be possible to remove all this codec work - just work through the fields of buildUpdate and send it straight up [{"op": "replace", "path": "/path/to/field", "value": *field}, ...]